src/control/conf.c

```diff
@@ -310,6 +310,23 @@ int dt_conf_get_bool(const char *name)
   return val;
 }
 
+gchar *dt_conf_get_path(const char *name)
+{
+  char *path = g_strdup(_conf_get_var(name));
+
+  if(!g_file_test(path, G_FILE_TEST_IS_DIR))
+  {
+    const dt_confgen_value_t *item = g_hash_table_lookup(darktable.conf->x_confgen, name);
+    if(item->type == DT_PATH)
+    {
+      g_free(path);
+      path = dt_conf_expand_default_dir(dt_confgen_get(name, DT_DEFAULT));
+    }
```

Comment:
Why not save that in the conf system right now?

Reply:
Done.